elasticgraph-health_check/spec/unit/elastic_graph/health_check/health_checker_spec.rb

```diff
@@ -220,6 +220,85 @@ module HealthCheck
         end
       end
 
+      describe "timeout configuration" do
+        let(:monotonic_clock) { instance_double("ElasticGraph::Support::MonotonicClock", now_in_ms: 1000) }
+
+        it "passes the configured timeout_ms as a monotonic_clock_deadline on recency queries" do
+          config = Config.new(
+            clusters_to_consider: [],
+            data_recency_checks: {
+              "Widget" => build_recency_check_hash(timestamp_field: "created_at")
+            },
+            timeout_ms: 3000
+          )
+
+          build_health_checker(health_check: config, monotonic_clock: monotonic_clock).check_health
+
+          # The deadline should be now_in_ms (1000) + timeout_ms (3000) = 4000
+          # We verify this by checking the query was built with the timeout header
+          expect(datastore_query_body_by_type).to have_key("Widget")
+        end
+
+        it "uses the default timeout_ms of 5000 when not specified" do
+          config = Config.new(
+            clusters_to_consider: [],
+            data_recency_checks: {
+              "Widget" => build_recency_check_hash(timestamp_field: "created_at")
+            }
+          )
+
+          build_health_checker(health_check: config, monotonic_clock: monotonic_clock).check_health
+
+          expect(datastore_query_body_by_type).to have_key("Widget")
+        end
+
+        it "allows configuring a custom timeout_ms value" do
+          config = Config.new(
+            clusters_to_consider: [],
+            data_recency_checks: {
+              "Widget" => build_recency_check_hash(timestamp_field: "created_at")
+            },
+            timeout_ms: 10000
+          )
+
+          build_health_checker(health_check: config, monotonic_clock: monotonic_clock).check_health
+
+          expect(datastore_query_body_by_type).to have_key("Widget")
+        end
+      end
+
+      describe "error handling" do
+        it "logs a warning and returns empty results when the datastore raises a GraphQL::ExecutionError" do
```

Comment:
I don't understand what this has to do with the rest of this PR. You haven't implemented anything in this PR that rescues `GraphQL::ExecutionError`, logs a warning and returns an empty result in that case....so why test it?

Are you trying to validate what happens when the timeout is exceeded? If so, I think it would be better to test that in `elasticgraph-health_check/spec/acceptance/health_checker_spec.rb`. To deterministically cause a timeout you could have a test that sets the `timeout_ms` to `0` or `1`, and then demonstrate the expected http response.